src/sentry/api/endpoints/user_notification_options.py

```diff
@@ -0,0 +1,61 @@
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.user import UserEndpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.api.serializers import serialize
+from sentry.api.validators.notifications import validate_type
+from sentry.models import User
+from sentry.models.notificationsettingoption import NotificationSettingOption
+from sentry.notifications.serializers import NotificationSettingsOptionSerializer
+from sentry.notifications.validators import UserNotificationSettingOptionWithValueSerializer
+
+
+@control_silo_endpoint
+class UserNotificationOptionsEndpoint(UserEndpoint):
+    # TODO(Steve): Make not private when we launch new system
+    private = True
+
+    def get(self, request: Request, user: User) -> Response:
```

Comment:
should you be gating this behind the feature flag you've added?

Reply:
@JoshFerge It's not needed and annoying to do. We have to query all the organizations for a user and check them individually. We will honor the settings once we do the migration anyways if someone used the API.